curve/curve.go

```diff
@@ -520,101 +521,81 @@ func (sc StarkCurve) SignFelt(msgHash, privKey *felt.Felt) (*felt.Felt, *felt.Fe
 //
-// The function appends the length of `elems` to the slice and then calls the `HashElements` method of the
-// `Curve` struct, passing in `elems` as an argument. The resulting hash and
-// any error that occurred during computation are returned.
+// The function appends the length of `elems` to the slice and then calls the `HashElements` method
+// passing in `elems` as an argument. The resulting hash is returned.
 //
 // Parameters:
 // - elems: slice of big.Int pointers to be hashed
 // Returns:
 // - hash: The hash of the list of elements
-// - err: An error if any
-func (sc StarkCurve) ComputeHashOnElements(elems []*big.Int) (hash *big.Int, err error) {
+func ComputeHashOnElements(elems []*big.Int) (hash *big.Int) {
 	elems = append(elems, big.NewInt(int64(len(elems))))
-	return Curve.HashElements((elems))
+	return HashElements(elems)
 }
 
-// PedersenHash calculates the Pedersen hash of the given elements.
-// NOTE: This function assumes the curve has been initialized with constant points
-// (ref: https://github.com/seanjameshan/starknet.js/blob/main/src/utils/ellipticCurve.ts)
-//
-// The function requires that the precomputed constant points have been initiated.
-// If the length of `sc.ConstantPoints` is zero, an error is returned.
-// The function iterates over the elements in `elems` and performs the Pedersen hash calculation.
-// For each element, it checks if the value is within the valid range.
-// If the value is invalid, an error is returned.
-// For each bit in the element, the function performs an addition operation on `ptx` and `pty`
-// using the corresponding constant point from the precomputed constant points.
-// If the constant point is a duplicate of `ptx`, an error is returned.
-// The function returns the resulting hash and a nil error if the calculation is successful.
-// Otherwise, it returns `ptx` and an error describing the issue encountered.
+// ComputeHashOnElementsFelt computes the hash on elements of a Felt array.
+// Does the same as ComputeHashOnElements, but receives and returns felt types.
 //
 // Parameters:
-// - elems: An array of big integers representing the elements to hash.
+// - feltArr: A pointer to an array of Felt objects.
 // Returns:
-// - hash: The resulting Pedersen hash as a big integer.
-// - err: An error, if any, encountered during the calculation.
-func (sc StarkCurve) PedersenHash(elems []*big.Int) (hash *big.Int, err error) {
-	if len(sc.ConstantPoints) == 0 {
-		return hash, fmt.Errorf("must initiate precomputed constant points")
-	}
-
-	ptx := new(big.Int).Set(sc.Gx)
-	pty := new(big.Int).Set(sc.Gy)
-	for i, elem := range elems {
-		x := new(big.Int).Set(elem)
-
-		if x.Cmp(big.NewInt(0)) == -1 || x.Cmp(sc.P) >= 0 {
-			return ptx, fmt.Errorf("invalid x: %v", x)
-		}
+// - *felt.Felt: a pointer to a Felt object
+func ComputeHashOnElementsFelt(feltArr []*felt.Felt) *felt.Felt {
+	return PedersenArray(feltArr...)
+}
```

Comment:
Can we remove this and just use PedersenArray?

Reply:
Yes, we can. This was an already existing function (moved from hash pkg), I left it there because there is a "ComputeHashOnElements" being used and a similar name could help the user; that's all.
Should we remove it anyway?

Reply:
Yeah let's remove it